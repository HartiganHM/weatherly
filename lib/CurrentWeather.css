.CurrentWeather {
	color: #FFF;
  display: flex;
  justify-content: space-between;
  border: 0.02rem solid ghostwhite;
  padding: 2rem;
  margin: auto;
}

.current-temp {
	font-size: 6rem;
}

.condition {
	font-size: 1.4rem;
}

h2 {
	font-size: 1.8rem;
}

.hi-lo {
	align-items: center;
	display: flex;
}

.spacer {
	margin: 0 2rem;
}

.weather-box {
	align-items: center;
	display: flex;
	flex: 1;
	flex-direction: column;
	justify-content: space-between;
}

.weather-icon {
	width: 15rem;
}