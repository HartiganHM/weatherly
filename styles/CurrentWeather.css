.CurrentWeather {
	color: #FFF;
	display: flex;
	justify-content: space-between;
	border: 0.02rem solid ghostwhite;
	padding: 2rem;
	margin: 1rem;
}

.currentHeader {
	color: #FFF;
	font-family: 'Open Sans';
	font-size: 1.8rem;
	padding: 0 0 1rem 1rem;
	text-align: left;
  }

.current-temp {
	font-size: 6rem;
}

.hi-lo {
	align-items: center;
	display: flex;
}

.rangeTemp {
	font-size: 3.2rem;
}

.subHeader {
	font-size: 1.2rem;
	margin: 0;
	padding: 0;
}

.spacer {
	margin: 0 2rem;
}

.weather-box {
	align-items: center;
	display: flex;
	flex: 1;
	flex-direction: column;
	justify-content: space-between;
}

.weather-icon {
	width: 15rem;
}