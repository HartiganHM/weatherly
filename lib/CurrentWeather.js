import React from 'react';
import '../styles/CurrentWeather.css';

const CurrentWeather = (props) => {
	const {city, icon, condition, day, date, tempF, tempC, tempHi, tempLo, summary } = props.data;
  return (
		<div>
			<h3 className='currentHeader'>Current Forecast</h3>
			<div className='CurrentWeather'>
				<div className='weather-box'>
					<h3 className='city'>{city}</h3>
					<img className='weather-icon' src={icon}/>
					<h4 className='subHeader'>{condition}</h4>
				</div>
				<div className='weather-box'>
				<h3>{day}</h3>
					<h3>{date}</h3>
					<h2 className='current-temp'>{tempF}째F</h2>
					<h4 className='subHeader'>{tempC}째C</h4>
				</div>
				<div className='weather-box'>
					<div className='hi-lo'><h3>Hi | Lo</h3></div>
					<div className='hi-lo'><h2 className='rangeTemp'>{tempHi}째</h2><span className='spacer'>/</span><h2 className='rangeTemp'>{tempLo}째</h2></div>
					<h4 className='subHeader'>{summary}</h4>
				</div>
			</div>
  	</div>
  )
}

export default CurrentWeather;