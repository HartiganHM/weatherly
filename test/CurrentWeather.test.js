import React from 'react';
import { shallow, mount } from 'enzyme';
import CurrentWeather from '../lib/CurrentWeather';

describe('CurrentWeather', () => {
    
    it('CurrentWeather should exist', () => {
        expect(<CurrentWeather />).toBeDefined()
    })

    it('Should have a className CurrentWeather', () => {
        const currentWeather = shallow(<div className='CurrentWeather'></div>);
        expect( currentWeather.find('.CurrentWeather').length ).toEqual(1);
    })
});