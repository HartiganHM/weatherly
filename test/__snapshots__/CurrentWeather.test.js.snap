// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather Should have a configurable city 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h3
        className="currentHeader"
    >
        Current Forecast
    </h3>
    <div
        className="CurrentWeather"
    >
        <div
            className="weather-box"
        >
            <h3
                className="city"
            >
                Denver
            </h3>
            <img
                className="weather-icon"
                src="http://icons.wxug.com/i/c/k/sunny.gif"
            />
            <h4
                className="subHeader"
            >
                Sunny
            </h4>
        </div>
        <div
            className="weather-box"
        >
            <h3>
                Friday
            </h3>
            <h3>
                November 3, 2017
            </h3>
            <h2
                className="current-temp"
            >
                32
                °F
            </h2>
            <h4
                className="subHeader"
            >
                0
                °C
            </h4>
        </div>
        <div
            className="weather-box"
        >
            <div
                className="hi-lo"
            >
                <h3>
                    Hi | Lo
                </h3>
            </div>
            <div
                className="hi-lo"
            >
                <h2
                    className="rangeTemp"
                >
                    212°F
                    °
                </h2>
                <span
                    className="spacer"
                >
                    /
                </span>
                <h2
                    className="rangeTemp"
                >
                    −459.67°F
                    °
                </h2>
            </div>
            <h4
                className="subHeader"
            >
                Sunny
            </h4>
        </div>
    </div>
</div>,
  "nodes": Array [
    <div>
      <h3
            className="currentHeader"
      >
            Current Forecast
      </h3>
      <div
            className="CurrentWeather"
      >
            <div
                  className="weather-box"
            >
                  <h3
                        className="city"
                  >
                        Denver
                  </h3>
                  <img
                        className="weather-icon"
                        src="http://icons.wxug.com/i/c/k/sunny.gif"
                  />
                  <h4
                        className="subHeader"
                  >
                        Sunny
                  </h4>
            </div>
            <div
                  className="weather-box"
            >
                  <h3>
                        Friday
                  </h3>
                  <h3>
                        November 3, 2017
                  </h3>
                  <h2
                        className="current-temp"
                  >
                        32
                        °F
                  </h2>
                  <h4
                        className="subHeader"
                  >
                        0
                        °C
                  </h4>
            </div>
            <div
                  className="weather-box"
            >
                  <div
                        className="hi-lo"
                  >
                        <h3>
                              Hi | Lo
                        </h3>
                  </div>
                  <div
                        className="hi-lo"
                  >
                        <h2
                              className="rangeTemp"
                        >
                              212°F
                              °
                        </h2>
                        <span
                              className="spacer"
                        >
                              /
                        </span>
                        <h2
                              className="rangeTemp"
                        >
                              −459.67°F
                              °
                        </h2>
                  </div>
                  <h4
                        className="subHeader"
                  >
                        Sunny
                  </h4>
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <CurrentWeather
        data={
                Object {
                        "city": "Denver",
                        "condition": "Sunny",
                        "date": "November 3, 2017",
                        "day": "Friday",
                        "icon": "http://icons.wxug.com/i/c/k/sunny.gif",
                        "summary": "Sunny",
                        "tempC": "0",
                        "tempF": "32",
                        "tempHi": "212°F",
                        "tempLo": "−459.67°F",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "data": Object {
            "city": "Denver",
            "condition": "Sunny",
            "date": "November 3, 2017",
            "day": "Friday",
            "icon": "http://icons.wxug.com/i/c/k/sunny.gif",
            "summary": "Sunny",
            "tempC": "0",
            "tempF": "32",
            "tempHi": "212°F",
            "tempLo": "−459.67°F",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h3
                    className="currentHeader"
          >
                    Current Forecast
          </h3>
          <div
                    className="CurrentWeather"
          >
                    <div
                              className="weather-box"
                    >
                              <h3
                                        className="city"
                              >
                                        Denver
                              </h3>
                              <img
                                        className="weather-icon"
                                        src="http://icons.wxug.com/i/c/k/sunny.gif"
                              />
                              <h4
                                        className="subHeader"
                              >
                                        Sunny
                              </h4>
                    </div>
                    <div
                              className="weather-box"
                    >
                              <h3>
                                        Friday
                              </h3>
                              <h3>
                                        November 3, 2017
                              </h3>
                              <h2
                                        className="current-temp"
                              >
                                        32
                                        °F
                              </h2>
                              <h4
                                        className="subHeader"
                              >
                                        0
                                        °C
                              </h4>
                    </div>
                    <div
                              className="weather-box"
                    >
                              <div
                                        className="hi-lo"
                              >
                                        <h3>
                                                  Hi | Lo
                                        </h3>
                              </div>
                              <div
                                        className="hi-lo"
                              >
                                        <h2
                                                  className="rangeTemp"
                                        >
                                                  212°F
                                                  °
                                        </h2>
                                        <span
                                                  className="spacer"
                                        >
                                                  /
                                        </span>
                                        <h2
                                                  className="rangeTemp"
                                        >
                                                  −459.67°F
                                                  °
                                        </h2>
                              </div>
                              <h4
                                        className="subHeader"
                              >
                                        Sunny
                              </h4>
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <h3
                    className="currentHeader"
          >
                    Current Forecast
          </h3>
          <div
                    className="CurrentWeather"
          >
                    <div
                              className="weather-box"
                    >
                              <h3
                                        className="city"
                              >
                                        Denver
                              </h3>
                              <img
                                        className="weather-icon"
                                        src="http://icons.wxug.com/i/c/k/sunny.gif"
                              />
                              <h4
                                        className="subHeader"
                              >
                                        Sunny
                              </h4>
                    </div>
                    <div
                              className="weather-box"
                    >
                              <h3>
                                        Friday
                              </h3>
                              <h3>
                                        November 3, 2017
                              </h3>
                              <h2
                                        className="current-temp"
                              >
                                        32
                                        °F
                              </h2>
                              <h4
                                        className="subHeader"
                              >
                                        0
                                        °C
                              </h4>
                    </div>
                    <div
                              className="weather-box"
                    >
                              <div
                                        className="hi-lo"
                              >
                                        <h3>
                                                  Hi | Lo
                                        </h3>
                              </div>
                              <div
                                        className="hi-lo"
                              >
                                        <h2
                                                  className="rangeTemp"
                                        >
                                                  212°F
                                                  °
                                        </h2>
                                        <span
                                                  className="spacer"
                                        >
                                                  /
                                        </span>
                                        <h2
                                                  className="rangeTemp"
                                        >
                                                  −459.67°F
                                                  °
                                        </h2>
                              </div>
                              <h4
                                        className="subHeader"
                              >
                                        Sunny
                              </h4>
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CurrentWeather
    data={
        Object {
            "city": "Denver",
            "condition": "Sunny",
            "date": "November 3, 2017",
            "day": "Friday",
            "icon": "http://icons.wxug.com/i/c/k/sunny.gif",
            "summary": "Sunny",
            "tempC": "0",
            "tempF": "32",
            "tempHi": "212°F",
            "tempLo": "−459.67°F",
          }
    }
/>,
}
`;
